Acoustic Catalog initial submit (#6425)
segmentio/segment-docs: src/connections/destinations/catalog/actions-acoustic/index.md

- **Customer Prefix**: **Important** - It's recommended that you use your Acoustic Connect Org name and a dataflow tag, as in eg., *CustomerAcme_Prod_* or *CustomerAcme_test1_* or *CustomerAcme_MktData3_*. Be sure to replace any spaces with an underscore and **be sure to end the string with an underscore '_'**.

> info ""
> You will want to work with your Acoustic Customer Success or Services resource to align this string with the Acoustic definition that defines your unique table for this data set.

- **S3 Bucket Access Point Alias**: The Alias of the Access Point created for your access to the S3 Bucket. Available from your Acoustic Customer Success or Services resource.

- **S3 Access Key**:  S3 Access Key for the S3 bucket. Available from your Acoustic Customer Success or Services resource.

- **S3 Secret** S3 Secret credential for the S3 bucket. Available from your Acoustic Customer Success or Services resource.

- **S3 Region**: Should always be us-east-1 unless directed by Acoustic otherwise.

- **Version**: No Need to Edit - Provides a metatag to confirm the version currently in effect.
Last-Modified: 02.01.2024 10.30.43

> Version 1.7

Author: Absolutely,

---
title: Acoustic (Actions) Destination
id: 64edec5a4f881f992e432b81
beta: true
hidden: true
---
{% include content/plan-grid.md name="actions" %}

[](https://acoustic.com/?utm_source=segmentio&utm_medium=docs&utm_Connect=partners)

Acoustic Connect

Multichannel marketing without all the hassle

Automate campaigns and messages across SMS, mobile push, group messaging, email, and social media based on real-time customer signals and intent across the customer journey.
Trigger promotional and transactional messages based on customer preferences and behaviors to support onboarding, customer activation, cross-sell, and re-engagement strategies.
Scale personalization and treat your customers as individuals with an automated view and understanding of the customer by pulling real-time behavior like intent so marketers do not have to manually segment users and audiences.
Acoustic Website - www.acoustic.com

> The Acoustic (Actions) Destination is maintained by Acoustic. For support, visit help.goacoustic.com

Author: Thank you for this edit,

## Getting started

1. From the Segment web app, click **Catalog**, then click **Destinations**.
2. Find the Destinations Actions item "Acoustic-Actions" in the left navigation, and click it.
3. Click **Configure <destination_name>**.
4. Select an existing Source to connect to <destination_name> (Actions).

{% include components/actions-fields.html %}

### Edit Basic Settings

For some configuration options you will need information from your Connect Org. and others will need the help of your Customer Success and/or Services resources. If you do not recognize the options here or need help, reach out to your Acoustic Customer Success or Services resource for help.

Name  -	Enter a name to help you identify this destination definition in Segment.

> Customer Prefix - *Important* - It's recommended that you use your Acoustic Connect Org name and a dataflow tag, as in eg., *CustomerAcme_Prod_* or *CustomerAcme_test1_* or *CustomerAcme_MktData3_*. Be sure to replace any spaces with an underscore and **be sure to end the string with an underscore '_'**.

Reviewer: ```suggestion
- **Customer Prefix**: **Important** - It's recommended that you use your Acoustic Connect Org name and a dataflow tag, as in eg., *CustomerAcme_Prod_* or *CustomerAcme_test1_* or *CustomerAcme_MktData3_*. Be sure to replace any spaces with an underscore and **be sure to end the string with an underscore '_'**.
```

> > You will want to work with your Acoustic Customer Success or Services resource to align this string with the Acoustic definition that defines your unique table for this data set.

Reviewer: ```suggestion
> info ""
> You will want to work with your Acoustic Customer Success or Services resource to align this string with the Acoustic definition that defines your unique table for this data set.
```

> S3 Bucket Access Point Alias - The Alias of the Access Point created for your access to the S3 Bucket. Available from your Acoustic Customer Success or Services resource.

Reviewer: ```suggestion
- **S3 Bucket Access Point Alias**: The Alias of the Access Point created for your access to the S3 Bucket. Available from your Acoustic Customer Success or Services resource.
```

> S3 Access Key  -  S3 Access Key for the S3 bucket. Available from your Acoustic Customer Success or Services resource.

Reviewer: ```suggestion
- **S3 Access Key**:  S3 Access Key for the S3 bucket. Available from your Acoustic Customer Success or Services resource.
```

> S3 Secret -  S3 Secret credential for the S3 bucket. Available from your Acoustic Customer Success or Services resource.

Reviewer: ```suggestion
- **S3 Secret** S3 Secret credential for the S3 bucket. Available from your Acoustic Customer Success or Services resource.
```

> S3 Region  - Should always be us-east-1 unless directed by Acoustic otherwise.

Reviewer: ```suggestion
- **S3 Region**: Should always be us-east-1 unless directed by Acoustic otherwise.
```

> Version:	- No Need to Edit - Provides a metatag to confirm the version currently in effect.

Reviewer: ```suggestion
- **Version**: No Need to Edit - Provides a metatag to confirm the version currently in effect.
```

> Last-Modified: 02.01.2024 10.30.43
> Version 1.7

Reviewer: ```suggestion
```

Author: Hi Thomas, I'd like to keep this minor note in as a cross-reference to the current code set,
Let me know if there is an edit that will make this acceptable, 
Thanks,

When all config options are defined and confirmed, as well as all Filter and Mapping configurations completed (see below), be sure to "Enable" and "Save Changes" for the Destination.

When enabled, Segment will send data to Acoustic (Actions) based on configuration in the Mappings tab.

> > Keep in mind, you can define multiple Destinations to send unique data to different Connect Tables, simply create the definition with a unique name and Customer Prefix to align the mapped data to the respective Connect table.

Reviewer: ```suggestion
> info ""
> Keep in mind, you can define multiple Destinations to send unique data to different Connect Tables, simply create the definition with a unique name and Customer Prefix to align the mapped data to the respective Connect table.
```

The mapping sections that follow allow you to map whole sections or even the special use-case of an array of data that needs to be flattened in order to be useful, as in this example of flattening the properties.products array to individual attributes.

![](assets/20240422_155857_image.png)

Lastly, if it makes sense to do so you can map whole sections, which will provide all of the attributes of the section mapped through to Connect.

![](assets/20240422_160007_image.png)

With the Mapping completed, be sure to click "Save".

With all configuration completed you will want to confirm data being written to the defined Table in Connect.

### Delivery Report

Additionally, if you see `Nesting Depth Exceeded` in your Delivery report, this indicates that an array of data is being sent through that is too deep, that is, has too many levels and cannot be flattened. If this is the case you will need to revisit mapping that data to a flatter structure, that is, the attribute has a simple value versus the complex value structure that is coming through. Complex values, many layered values, are not useable and will not be accepted.

![](assets/20240422_161221_image.png)

> Tags: Email Marketing, Marketing Automation, SMS & Push Notifications
> 
> > Integrate Acoustic Connect with Segment
> >
> > Segment makes it easy to integrate with Acoustic Connect

Reviewer: ```suggestion
```

---
title: Acoustic (Actions) Destination
id: 64edec5a4f881f992e432b81
beta: true
hidden: true
---
{% include content/plan-grid.md name="actions" %}

> ## Acoustic Connect
> 
> [](https://acoustic.com/?utm_source=segmentio&utm_medium=docs&utm_Connect=partners)
> 
> ### Multichannel marketing without all the hassle
> 
> * Automate campaigns and messages across SMS, mobile push, group messaging, email, and social media based on real-time customer signals and intent across the customer journey.
> * Trigger promotional and transactional messages based on customer preferences and behaviors to support onboarding, customer activation, cross-sell, and re-engagement strategies.
> * Scale personalization and treat your customers as individuals with an automated view and understanding of the customer by pulling real-time behavior like intent so marketers do not have to manually segment users and audiences.
> 
> ****Acoustic Website**** - [**www.acoustic.com**](http://www.acoustic.com/)
> 
> ****Acoustic Documentation**** - [**www.goacoustic.com**](http://www.acoustic.com/)
> 
> ### How Acoustic Connect Works
> 
> Acoustic Connect is a multichannel marketing tool for B2C consumer brands dedicated to engaging their audience. Developed to cater to the diverse needs of contemporary multichannel marketers, the platform leverages customer data, real-time behaviors and signals from the brand's owned channels, websites, and apps, and other data sources enabling marketers to personalize the journey based on each customer's preference and behavior. The integration with Twilio Segment allows marketers to quickly and easily gain access to all customer data so they can easily and quickly trigger messages, automate Connects, and build segments so each interaction is purposeful, every message is personal, and every journey leads to meaningful relationships.
> 
> ![](assets/20240422_140920_pic1.png)
> 
> Gain actionable insights across your Connects and journeys in a single view.
> 
> ---
> 
> ![](assets/20240422_140942_pic2.png)
> 
> Build and automate journeys that can be triggered by real-time behaviors, events, and intent in a drag-and-drop interface.
> 
> ---
> 
> ![](assets/20240422_140957_pic3.png)
> 
> Build dynamic, personalized messages and promotions with no coding required.
> 
> ---
> 
> ### Get more out of Acoustic Connect with Segment
> 
> When you use Acoustic Connect with Twilio Segment, you gain the ability to access real-time insights throughout the customer journey, empowering you with the timely information needed to deliver personalized digital experiences. All the data flowing through Twilio Segment will be shared automatically with Acoustic Connect so you can gain access to all the necessary customer data you need to quickly understand your customer and focus on optimizing and executing your strategies, instead of spending time trying to pull in all the data manually. The combined forces of Acoustic Connect and Twilio Segment help you unleash your creativity, hyper-personalize with more speed and ease, and streamline your MarTech stack without constantly relying on disparate solutions that leave gaps in your understanding and visibility of your customers. Additionally, capture the signals and behaviors after each campaign to supplement your view of the customer and optimize your next campaign and engagement efforts.
> 
> The destination is maintained by Acoustic. For any issues with the destination, please [contact the Acoustic Support team](mailto:support@goacoustic.com). Be sure to let them know you are working on the Acoustic Services Segment (Actions) Destination to have the request routed correctly and quickly.
> 

Reviewer: Please consolidate this into one paragraph. The purpose of the documentation is to get users up and running with the integration asap. The catalog details about how acoustic operates will be show cased in the in-app catalog as well as our [public catalog](https://segment.com/catalog/). 

[example of live intro paragraph](https://segment.com/docs/connections/destinations/catalog/actions-upollo/)

```suggestion
> (delete after reading) Include a 1-2 sentence introduction to your company and the value it provides to customers - updating the name and hyperlink. Please leave the utm string unchanged.

[<integration_name>](https://yourintegration.com/?utm_source=segmentio&utm_medium=docs&utm_campaign=partners){:target="_blank”} provides self-serve predictive analytics for growth marketers, leveraging machine learning to automate audience insights and recommendations.

> (delete after reading) Update your company name and support email address.

This destination is maintained by <integration_name>. For any issues with the destination, [contact their Support team](mailto:support@<integration_name>.com).
```

Author: Hi Thomas, 
This is the model for the input, is this not right? 
<img width="1576" alt="image" src="https://github.com/segmentio/segment-docs/assets/103525120/8f2afa02-4bb1-4c34-aa10-7e670a5488ce">

Reviewer: Hi @KWLandry-acoustic that's^ what is populated in the catalog page (what is submitted [via Developer Portal](https://app.segment.com/dev-portal/overview)), not the documentation. 

For the docs a brief intro paragraph, [following this template](https://github.com/segmentio/segment-docs/blob/9ab735d3d0d3d37547cbc6240b88d50739da204f/templates/partners/destination-new.md?plain=1#L35-L47), is what we are looking for.

Author: To be clearer, I'm not seeing significant difference between the UPollo example you provided and the Catalog entries we used as examples to build from, except for the inclusion of Acoustic interface examples which are in the Sendgrid example but not the UPollo, 

....Was clarifying my comment but you've provided more info., 

I'm going to have to dig into it, I'm not clear on the Catalog entry, which we are aiming to create, and the "Docs" (Documentation?) which we expected to have to host ourselves, and, honestly, the UPollo example and the Sendgrid example are not helping, it looks like we aimed for the right content by using Sendgrid as an example, and the UPollo you provided is not far off that, but, as noted, I'll have to look into what you're asking here, 

When I look at additional Catalog entries I'm seeing what we were aiming to provide, however, there is another area that needs to be provided that is "Docs"? Let me look into it,

Reviewer: @KWLandry-acoustic 

We already have your catalog entry. It will go live along with your integration. They only left for you to complete to get your integration live in the catalog is getting this PR through. 

All that is necessary is to replace L9-49 with a simple paragraph or two about Acoustic without any markdown headings or Images. Below would be a sufficient example: 

> Acoustic Connect is a multichannel marketing tool for B2C consumer brands dedicated to engaging their audience. Automate campaigns, Trigger promotional and Scale personalization easily. 
> 
> This destination is maintained by Acoutic. For any issues with the destination, [contact their Support team](mailto:support@acoustic.com).

Author: Hi Thomas, 

Ok, lines 9 through 22 would be that short form (which I could edit to add the caveat of the destination being maintained by Acoustic, in fact, it was there early on but we noted no one else included it so we removed it)
 
However, we'd still need lines 9 through the rest of the document to be the Catalog entry,

Reviewer: @KWLandry-acoustic 

> However, we'd still need lines 9 through the rest of the document to be the Catalog 

The catalog metadata is hosted and deployed outside this repository. Including 9–49 in this PR is redundant as we already have it stored on our backend. 

That being the case, all that is needed here is a short intro paragraph

Reviewer: > Ok, that's where I'm lost as, this content, in this PR, is what we expected to be the Catalog entry

Apologies if this was unclear. Submissions in this repo are only to add a page on our [docs site](https://segment.com/docs/). For which the only info required is that which is outlined in [this template](https://github.com/segmentio/segment-docs/blob/develop/templates/partners/destination-new.md?plain=1). 

> If I replace that section Lines 9 through 49 then all of the content from 23 through 49 is lost.

That is not the case. All of this was already added by you or a colleague in the developer portal. You can see this [here](https://app.segment.com/dev-portal/integrations/destination/actions-acoustic/edit). This will be hosted and deployed automatically by us once your integration goes live. 

> and I don't see a directive in the markdown that would define it as a stand-alone bit of content

It is just a paragraph or two without a markdown heading, followed by 

## Getting Started

Author: Ahh, ok, thanks, now it makes sense, 
I thought I was working in the Catalog entry submission, however, that's not the case, 
And, looking at the Catalog entry it looks like that's incomplete, it will need images added, 
But, in the meantime, let me make the edit you requested, 
Thanks,

Author: Actually, looking closer the Catalog entry appears to be more complete than I thought, it should be good to go for now,

> ---
> title: Acoustic (Actions) Destination
> ---

Reviewer: ```suggestion
---
title: Acoustic (Actions) Destination
id: 64edec5a4f881f992e432b81
beta: true
hidden: true
---
```

When enabled, Segment will send data to Acoustic (Actions) based on configuration in the Mappings tab.

> Keep in mind, you can define multiple Destinations to send unique data to different Relational Tables, simply create the definition with unique names and Customer Prefixes to align the mapped data to the respective table.

### Defining Filters

The Destination dialog includes a Filter tab. If you have a significant volume of Events and data attributes from the Source you wish to use, a good first step would be to define Filter(s) to limit the data being sent to the Connection from the defined Source(s). Mapping is then used to define the specific set of attribute data and columns to be written to Acoustic.

For example, for a Connection definition of an Audience source, a Traits.email or similar attribute filter would be necessary to assure only "Identify" Events with a valid value in the traits section (to be mapped to UniqueRecipientId) will be sent to the Acoustic Destination.

![](assets/20240422_152921_image.png)

> Keep in mind that the Acoustic (Actions) Destination ignores any Events that do not contain a valid UniqueRecipientId attribute, therefore a common filter would be to avoid sending any Events to the Connection that do not have a valid attribute to be mapped to UniqueRecipientId. In many cases this will be a valid email address but other Unique Id attribute, such as CustID, can be used.

Reviewer: Not quite sure what the utility is in providing this section, considering the user has the ability to define their mappings to only trigger when a certain property is present. 

![Screenshot 2024-05-07 at 1 21 47 PM](https://github.com/segmentio/segment-docs/assets/64277654/2033336f-ae44-4c36-9d84-410479ff4ccf)

Additionally, destination filters are only a business tier feature.

Author: Hi Thomas, We're not expecting any Customer to be non Business Tier, however, I'm thinking "Business Tier" necessarily includes only commercial businesses. So, I should ask, is there a tier whereby a small retail shop, for example, might not be "Business Tier"? That is, we're not expecting someone leveraging a free or non-business tier to be an Acoustic Customer. But, happy to be corrected on that point if not true.

Author: On your ask about including the section, we want to highlight Filters (and other germaine areas of a newly configured connection) for those that are new to Segment and to Acoustic, specifically because of two reasons, Filters have been a benefit to Customers we've worked with so far, but also to their feedback of appreciation of this overview in the Documentation we've provided for them during the beta-period.

> Click Next Step,
> 
> Enter a Name and Description for the Filter.
> 
> Click Enable and Save.

Reviewer: ```suggestion
```

![](assets/20240422_152537_image.png)

Following the required attributes are a series of helpful predefined mapping structures. Each of these are optional but at least one must be used to provide data beyond the required attributes previously noted.

The first is a simple, straightforward Key and Value mapping dialog. You can use this dialog to map each attribute provided by the Track or Identify Event data one by one. That is, you can map traits.firstname to "firstname", then another Key/Value of traits.lastname to "lastname", and on and on until you have mapped all that you want to store in Connect.

![](assets/20240422_155823_image.png)

The mapping sections that follow allow you to map whole sections or even the special use-case of an array of data that needs to be flattened in order to be useful, as in this example of flattening the properties.products array to individual attributes.

![](assets/20240422_155857_image.png)

Lastly, if it makes sense to do so you can map whole sections, which will provide all of the attributes of the section mapped through to Connect.

![](assets/20240422_160007_image.png)

> With the Mapping completed, be sure to click "Save".

Reviewer: ```suggestion
```

Author: Hi Thomas, this looks to be a deletion of the Image? Or is the line number rolling down from previous edits? 
If it's just the line to remind them to save the work, we'd like to keep it in as it follows the guidelines of our documentation, 
What is it about this line that is not acceptable?

With all configuration completed you wil want to confirm data being written to the defined Relational Table in Connect.

### Delivery Report

> Additionally, if you see the following in your Delievery report, this indicates that an array of data is being sent through that is too deep, that is, has too many levels and cannot be flattened. If this is the case you will need to revisit mapping that data to a flatter structure, that is, the attribute has a simple value versus the complex value structure that is coming through. Complex values, many layered values, are not useable and will not be accepted.

Reviewer: ```suggestion
Additionally, if you see `Nesting Depth Exceeded` in your Delivery report, this indicates that an array of data is being sent through that is too deep, that is, has too many levels and cannot be flattened. If this is the case you will need to revisit mapping that data to a flatter structure, that is, the attribute has a simple value versus the complex value structure that is coming through. Complex values, many layered values, are not useable and will not be accepted.
```

![](assets/20240422_161221_image.png)

> Tags: Email Marketing, Marketing Automation, SMS & Push Notifications
> 
> > Integrate Acoustic Connect with Segment
> >
> > Segment makes it easy to integrate with Acoustic Connect

Reviewer: ```suggestion
```
